packages/fx-core/src/core/tools.ts

```diff
@@ -118,6 +119,24 @@ export function isValidProject(workspacePath?: string): boolean {
   }
 }
 
+export function getActiveEnv(projectRoot: string): string | undefined {
+  if (!isMultiEnvEnabled()) {
+    return "default";
+  }
+  try {
+    if (isValidProject(projectRoot)) {
+      const settingsJsonPath = path.join(
+        projectRoot,
+        `.${ConfigFolderName}/${InputConfigsFolderName}/${ProjectSettingsFileName}`
+      );
+      const settingsJson = JSON.parse(fs.readFileSync(settingsJsonPath, "utf8"));
+      return settingsJson.activeEnvironment;
+    }
+  } catch (e) {
+    return undefined;
```

Comment:
should we handle the error like `InvalidJson` or `activeEnvironment` is not set in projectSettings.json?

Reply:
`InvalidJson` and other IO errors are handled and returns undefined. If activeEnvironment is not set, it will return undefined. But I think it makes sense to check whether `activeEnvironment` is a string

Reply:
We need to report accurate user error, sometimes it would be `InvalidJson` and sometimes it would be `activeEnvironment is missing`. My point the return value of `undefined` cannot help us distinguish above cases.

Reply:
It makes sense. I will let the function to return `Promise<Result<string, FxError>>`